### PR TITLE
processor/sinkmanager(ticdc): background GC sort engine data

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflow/issues/5928

See https://github.com/pingcap/tiflow/issues/5928#issuecomment-1318253453

### What is changed and how it works?
Background GC sort engine data.

#### Tests 

 - [x] Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/processor/sinkmanager/manager.go [hunk 1]
```diff
@@ -247,6 +259,39 @@ func (m *SinkManager) startGenerateTasks() {
...
+		defer m.wg.Done()
+		for {
+			select {
+			case <-m.ctx.Done():
+				log.Info("Background GC is stooped because context is canceled",
+					zap.String("namespace", m.changefeedID.Namespace),
+					zap.String("changefeed", m.changefeedID.ID))
+				return
+			case gcEvent := <-m.engineGCChan:
+				if err := m.sortEngine.CleanByTable(gcEvent.tableID, gcEvent.cleanPos); err != nil {
```
Comment (nongfushanquan) [inline cdc/processor/sinkmanager/manager.go:275]: Is this method async?

Comment (hi-rustin) [reply cdc/processor/sinkmanager/manager.go:275]: Yes

**Review by nongfushanquan — APPROVED** [review]

File: cdc/processor/sinkmanager/manager.go [hunk 2]
```diff
@@ -596,9 +641,18 @@ func (m *SinkManager) GetTableStats(tableID model.TableID) (pipeline.Stats, erro
...
-	if err != nil {
-		return pipeline.Stats{}, errors.Trace(err)
+	gcEvent := &gcEvent{
+		tableID:  tableID,
+		cleanPos: cleanPos,
+	}
+	select {
+	case m.engineGCChan <- gcEvent:
+	default:
+		log.Debug("Failed to send GC event to engine GC channel, engine GC channel is full",
```
Comment (hicqu) [inline cdc/processor/sinkmanager/manager.go:651]: How about change the log level to `Warn`?

**Comment by hicqu** [comment]
Rest LGTM.

File: cdc/processor/sinkmanager/manager.go (hunk 2 above)
Comment (hi-rustin) [reply cdc/processor/sinkmanager/manager.go:651]: I guess it will be too annoying.

Comment (hicqu) [reply cdc/processor/sinkmanager/manager.go:651]: If they are dropped, we must have some ways to know it and then we can fix it.

**Review by hicqu — APPROVED** [review]

**Pull request merged**